specs/Picker-API_spec.md

```diff
@@ -0,0 +1,115 @@
+- Windows.Storage.Pickers.FileSavePicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.filesavepicker)
+- Windows.Storage.Pickers.FolderPicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.folderpicker) 
+
+# Description
+
+The goal of this API is to address the gaps in the existing File/Folder picker APIs. The API surface is similar to the existing
+Picker APIs in Windows SDK with additional methods to support File + Folder picker and multiple folder picker. The API removes the
+deprecated methods from Windows SDK.
+
+# Examples
+
+## Show file and folder picker
+    ```c#
+    var picker = StorageItemPicker::CreateStorageItemPicker();
+    picker.Filters.add("*.jpg");
+    picker.Filters.add("*.png");
+    picker.StartLocation = startLocation;
+    picker.PickerMode = StorageItemPickerMode::Any;
+    IReadOnlyList<StorageItem> storageItems = await picker.PickStorageItemAsync();
+
+    if(0 < storageItems.Count)
+    {
+    }
+    ```
+
+## Show single folder picker
+    ```c#
+    var picker = StorageItemPicker::CreateStorageItemPicker();
+    picker.PickerMode = StorageItemPickerMode::Folder;
+    Windows.Storage.StorageFolder folder = await picker.PickSingleFolderAsync();
+    ```
+
+
+## Show multi folder picker
+    ```c#
+    var picker = StorageItemPicker::CreateStorageItemPicker();
+    picker.PickerMode = StoargeItemPickerMode::Folder;
+    IReadOnlyList<Windows.Storage.StorageFolder> folders = picker.PickMultipleFoldersAsync();
+    ```
+
+# Remarks
+
+# API Notes
+
+
+# API Details
+
+    ```c#
+    namespace Microsoft.Reunion.Storage.Pickers
+    {
+        typedef enum PickerViewMode
+        {
+            List = 0,
+            Details,
+            Tiles,
+            Content
+        } PickerViewMode;
+
+        typedef enum StorageItemPickerMode
+        {
+            Any = 0,
+            File,
+            Folder
+        } StorageItemPickerMode;
+
+        runtimeclass StorageItemPicker
+        {
+            /// Static constructor
+            static StorageItemPicker StorageItemPicker();
+
+            /// Properties
+            String StartLocation{ set; }
+            String CommitButtonText { set; }
+            String CancelButtonText { set; }
+            Windows.Foundation.Collections.IMap<String, String> FilterAndDescription{ get; }
+            PickerViewMode ViewMode{ get; set; }
+            StorageItemPickerMode PickerMode { get; set; }
+            String SuggestedSaveFile{ get; set; }
+            String DefaultFileExtension{ get; set; }
+
+            /// Methods
+            Windows.Foundation.IAsyncOperation<Windows.Storage.StorageFile> PickSingleFileAsync();
+            Windows.Foundation.IAsyncOperation<Windows.Foundation.Collections.IVectorView<Windows.Storage.StorageFile>> PickMutipleFilesAsync();
+            Windows.Foundation.IAsyncOperation<Windows.Foundation.Collections.IVectorView<Windows.Storage.StorageItem>> PickStorageItemsAsync();
```

Comment:
If we're going to have the `Pick{specific type}Async` methods here, do we need the "mode" above indicating files vs. folders? Can we either:

* Remove the mode above and infer it from the method called (not great because that means N overloads)
* Keep the mode above but consolidate these methods down to PickSingleItemAsync/PickMultipleItemsAsync here?

Reply:
Consolidated the API to a single ShowPickerAsync. The user can set the PickerType using the properties.